mod/node-api/backend/types.go

```diff
@@ -117,19 +119,16 @@ type StorageBackend[
 	StateFromContext(context.Context) BeaconStateT
 }
 
-// Validator represents an interface for a validator with generic withdrawal
-// credentials. WithdrawalCredentialsT is a type parameter that must implement
-// the WithdrawalCredentials interface.
-type Validator[WithdrawalCredentialsT WithdrawalCredentials] interface {
-	// GetWithdrawalCredentials returns the withdrawal credentials of the
-	// validator.
-	GetWithdrawalCredentials() WithdrawalCredentialsT
-	// IsFullyWithdrawable checks if the validator is fully withdrawable given a
-	// certain Gwei amount and epoch.
-	IsFullyWithdrawable(amount math.Gwei, epoch math.Epoch) bool
-	// IsPartiallyWithdrawable checks if the validator is partially withdrawable
-	// given two Gwei amounts.
-	IsPartiallyWithdrawable(amount1 math.Gwei, amount2 math.Gwei) bool
+// Validator represents an interface for a validator.
+type Validator interface {
+	GetWithdrawalCredentials() types.WithdrawalCredentials
```

Comment:
re-add godoc, also WithdrawlCredentials should be a generic

Reply:
Added comments on the functions. Made WithdrawalCredentialsT of type WithdrawalCredentials.

Reply:
validator struct defn should be `Validator[WithdrawalCredentialsT WithdrawalCredentials]` still